modin/pandas/test/dataframe/test_binary.py

```diff
@@ -167,28 +167,68 @@ def test_multi_level_comparison(data, op):
         getattr(modin_df_multi_level, op)(modin_df_multi_level, axis=0, level=1)
 
 
-def test_equals():
-    frame_data = {"col1": [2.9, 3, 3, 3], "col2": [2, 3, 4, 1]}
-    modin_df1 = pd.DataFrame(frame_data)
-    modin_df2 = pd.DataFrame(frame_data)
-
-    assert modin_df1.equals(modin_df2)
-
-    df_equals(modin_df1, modin_df2)
-    df_equals(modin_df1, pd.DataFrame(modin_df1))
-
-    frame_data = {"col1": [2.9, 3, 3, 3], "col2": [2, 3, 5, 1]}
-    modin_df3 = pd.DataFrame(frame_data, index=list("abcd"))
-
-    assert not modin_df1.equals(modin_df3)
-
-    with pytest.raises(AssertionError):
-        df_equals(modin_df3, modin_df1)
-
-    with pytest.raises(AssertionError):
-        df_equals(modin_df3, modin_df2)
+@pytest.mark.parametrize(
+    "frame1_data,frame2_data,expected_pandas_equals",
+    [
+        pytest.param({}, {}, True, id="two_empty_dataframes"),
+        pytest.param([[1]], [[0]], False, id="single_unequal_values"),
+        pytest.param([[None]], [[None]], True, id="single_none_values"),
+        pytest.param({1: [10]}, {1.0: [10]}, True, id="different_column_types"),
+        pytest.param({1: [10]}, {2: [10]}, False, id="different_columns"),
+        pytest.param(
+            pandas.DataFrame({1: [10]}, index=[1]),
+            pandas.DataFrame({1: [10]}, index=[1.0]),
+            True,
+            id="different_index_types",
+        ),
+        pytest.param(
+            pandas.DataFrame({1: [10]}, index=[1]),
+            pandas.DataFrame({1: [10]}, index=[2]),
+            False,
+            id="different_indexes",
+        ),
+        pytest.param({1: [10]}, {1: [10.0]}, False, id="different_value_types"),
+        pytest.param(
+            [[1, 2], [3, 4]],
+            [[1, 2], [3, 4]],
+            True,
+            id="equal_two_by_two_dataframes",
+        ),
+        pytest.param(
+            [[1, 2], [3, 4]],
+            [[5, 2], [3, 4]],
+            False,
+            id="unequal_two_by_two_dataframes",
+        ),
+        pytest.param(
+            [[1, 1]],
+            [[1]],
+            False,
+            id="different_row_lengths",
+        ),
+        pytest.param(
+            [[1], [1]],
+            [[1]],
+            False,
+            id="different_column_lengths",
+        ),
+    ],
+)
+def test_equals(frame1_data, frame2_data, expected_pandas_equals):
+    modin_df1 = pd.DataFrame(frame1_data)
+    pandas_df1 = pandas.DataFrame(frame1_data)
+    modin_df2 = pd.DataFrame(frame2_data)
+    pandas_df2 = pandas.DataFrame(frame2_data)
+
+    pandas_equals = pandas_df1.equals(pandas_df2)
+    assert pandas_equals == expected_pandas_equals, (
+        f"Test expected pandas to say the dataframes were"
+        f"{'' if expected_pandas_equals else ' not'} equal, but they were"
+        f"{' not' if expected_pandas_equals else ''} equal."
+    )
 
-    assert modin_df1.equals(modin_df2._query_compiler.to_pandas())
+    assert modin_df1.equals(modin_df2) == pandas_equals
```

Comment:
nit: would it be useful to have a test case for `modin_df1.equals(modin_df1)` given that the original bug is that a DF is not `.equals(` to itself.

Reply:
The author of the bug just used that to show that to show that there was definitely a bug in Modin, because any object should be equal to itself. The root cause was that our `equals` doesn't work correctly when the dataframes include `None`. I think it's clearer to have the tests compare dataframes that should be equivalent because they were constructed from the same data.